src/compiler/checker.ts

```diff
@@ -3883,6 +3883,19 @@ namespace ts {
             return result || emptyArray;
         }
 
+        function getNamedOrIndexSignatureMembers(members: SymbolTable): Symbol[] {
+            const result = getNamedMembers(members);
+            const index = getIndexSymbolFromSymbolTable(members);
+            if (!index) {
+                return result;
+            }
+            if (result === emptyArray) {
+                return [index];
+            }
+            result.push(index);

```

Comment:
Is this... Safe? Is `getNamedMembers` guaranteed to return a new array every call that it doesn't return `emptyArray`? Does it not return a cached result? If it returns a unique array every call, this is fine (maybe a comment noting the required invariant would be useful). If not, we probably would need to guarantee a unique, new result array here.